hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/dstable/DistributionTable.java

```diff
@@ -573,9 +567,9 @@ private Object getPinButton(final Object itemId) {
     }
 
     private void saveDistributionPinnedBtn(final Button pinBtn) {
-        if (managementUIState.getTargetTableFilters().getPinnedDistId().isPresent()
-                && managementUIState.getTargetTableFilters().getPinnedDistId().get()
-                        .equals(((DistributionSetIdName) pinBtn.getData()).getId())) {
+        final Long pinnedId = ((DistributionSetIdName) pinBtn.getData()).getId();
```

Comment:
Maybe this can throw a NPE, if the `managementUIState.getTargetTableFilters().getPinnedDistId()` is not present.  Have to check when the pintBtn sets the data